toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -843,21 +861,27 @@ func addUsers(installChroot *safechroot.Chroot, users []configuration.User) (err
 	// If no root entry was specified in the config file, never expire the root password
 	if !rootUserAdded {
 		logger.Log.Debugf("No root user entry found in config file. Setting root password to never expire.")
+
+		// Ignore updating if there is no shadow file to update
+		if exists, _ := file.PathExists(shadowFile); !exists {
+			logger.Log.Debugf("No shadow file to update. Skipping.")
```

Comment:
It might be worth to say what we're skipping. It'll also make it easier to find debug the code from the logs if each logs is unique. Right now we're printing the same thing in the function below.